great_expectations/experimental/datasources/file_path_data_asset.py

```diff
@@ -47,7 +46,14 @@
 class _FilePathDataAsset(DataAsset):
     _EXCLUDE_FROM_READER_OPTIONS: ClassVar[Set[str]] = {
         "name",
-        "regex",
+        "regex",  # file_path argument
+        "base_directory",  # filesystem argument
```

Comment:
Based on discussions with @Kilo59 we determined that we can avoid combinatorial enumeration of data assets for every storage system by sharing the settings between the `Datasource` and the `add_<type>_asset()` method, such that the `DataConnector` for the `DataAsset` could be instantiated in order to provide (via the top-level `DataConnector` interface methods) all utilities needed by `DataAsset` to access files on different storage environments.  As a consequence, the exclusions in `FilePathDataAsset` have to account for the settings of all supported storage environments.  In addition, top-level `Datasource.add_asset()` method must accept additional arguments (such as instantiated `DataConnector`).

Reply:
I don't undestand why this is necessary. All these are arguments to the injected `DataConnector`, they don't need to be fields/attributes on the `DataAsset`.

Reply:
@Kilo59 I placed those exclusions with comments, to prevent such strange errors as `ExecutionEngine.__init__()` is not aware of argument “glob_directive” (I am paraphrasing).  If there is a better way to address this issue?  I cleaned up as much as I could (with now a better understanding with how we are using `Pydantic` -- and it concerns me a little but that we use `extra = pydantic.Extra.allow` here and there, which defeats the premise of strong validation; however, we can defer any actions regarding this for now).

Reply:
This issue is that these should not be fields/attrs on the `DataAsset` at all. They only needed by the `DataConnector`.

Reply:
@Kilo59 As of now, we have solved the problem, subject to a potential refactor (as per https://github.com/great-expectations/great_expectations/pull/7197#discussion_r1116098706).  Thanks.